src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -1216,6 +1218,127 @@ class CopyOptionsSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment(replace=True)
+class CreateSchemaStatementSegment(BaseSegment):
+    """A `CREATE SCHEMA` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-schema.html
+    """
+
+    type = "create_schema_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        Ref("TemporaryTransientGrammar", optional=True),
+        "SCHEMA",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("SchemaReferenceSegment"),
+        Sequence("WITH", "MANAGED", "ACCESS", optional=True),
+        AnyNumberOf(Ref("SchemaObjectParamsSegment")),
+        Sequence(
+            Sequence("WITH", optional=True),
+            "TAG",
+            Bracketed(
+                Delimited(
+                    Ref("NakedIdentifierSegment"),
+                    Ref("EqualsSegment"),
+                    Ref("QuotedIdentifierSegment"),
+                )
+            ),
+            optional=True,
+        ),
+    )
+
+
+@snowflake_dialect.segment()
+class AlterSchemaStatementSegment(BaseSegment):
+    """An `ALTER SCHEMA` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/alter-schema.html
+
+    """
+
+    type = "alter_schema_statement"
+    match_grammar = Sequence(
+        "ALTER",
+        "SCHEMA",
+        Sequence("IF", "EXISTS", optional=True),
+        Ref("SchemaReferenceSegment"),
+        OneOf(
+            Sequence(
+                "RENAME",
+                "TO",
+                Ref("SchemaReferenceSegment"),
+            ),
+            Sequence(
+                "SWAP",
+                "WITH",
+                Ref("SchemaReferenceSegment"),
+            ),
+            Sequence(
+                "SET",
+                AnyNumberOf(
+                    Ref("SchemaObjectParamsSegment"),
+                ),
+            ),
+            Sequence(
+                "UNSET",
+                AnyNumberOf(
+                    Delimited(
+                        "DATA_RETENTION_TIME_IN_DAYS",
+                        "MAX_DATA_EXTENSION_TIME_IN_DAYS",
+                        "DEFAULT_DDL_COLLATION",
+                        "COMMENT",
+                    ),
+                    Sequence("TAG", Delimited(Ref("NakedIdentifierSegment"))),
```

Comment:
This sequence should be in the Delimited above

Reply:
Grabbing the errors for this comment also made me run combinations of the unset command. Turns out that `unset tag` cannot be combined with any of the other options, so we need a `OneOf()` here.

Reply:
Grabbing the error codes for [this comment](https://github.com/sqlfluff/sqlfluff/pull/1950#discussion_r757817478) made me try variations of the unset command as well. It turns out that `unset tag` cannot be combined with any of the other options, so we need a `OneOf()` here.

Reply:
Okay!